geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateDefinedIndexesCommand.java

```diff
@@ -59,77 +58,27 @@ public ResultModel createDefinedIndexes(
     ResultModel result = new ResultModel();
 
     if (IndexDefinition.indexDefinitions.isEmpty()) {
-      result.addInfo().addLine(CliStrings.DEFINE_INDEX__FAILURE__MSG);
-      return result;
+      return ResultModel.createInfo(CliStrings.DEFINE_INDEX__FAILURE__MSG);
     }
 
     Set<DistributedMember> targetMembers = findMembers(groups, memberNameOrID);
     if (targetMembers.isEmpty()) {
-      result.addInfo().addLine(CliStrings.NO_MEMBERS_FOUND_MESSAGE);
-      result.setStatus(Result.Status.ERROR);
-      return result;
+      return ResultModel.createError(CliStrings.NO_MEMBERS_FOUND_MESSAGE);
     }
 
-    // This command is very horrible because there is no definitive correlation between where
-    // indexes might have been created and where errors occurred. Thus it may be very difficult to
-    // determine what to update for the cluster configuration. We try to alleviate this problem by
-    // applying changes iteratively.
-    // It may be better to limit this command to EITHER executing on a list of members OR
-    // to execute on a SINGLE group.
-
-    // First let's do the explicitly provided members. This implies changes to the whole cluster.
-    // However this is also not always a correct assumption as the user could be creating an index
-    // on a REPLICATED region on a single member. (Indexes created on PARTITIONED regions are
-    // automatically distributed).
-
-    Result.Status status;
-    TabularResultModel table = result.addTable(CREATE_DEFINED_INDEXES_SECTION);
     ResultCollector<?, ?> rc = executeFunction(createDefinedIndexesFunction,
         IndexDefinition.indexDefinitions, targetMembers);
 
-    status = accumulateResults(table, (List<CliFunctionResult>) rc.getResult());
-    if (status == Result.Status.ERROR) {
-      result.setStatus(status);
-    } else {
+    result.addTableAndSetStatus(CREATE_DEFINED_INDEXES_SECTION,
+        (List<CliFunctionResult>) rc.getResult(), false);
+
+    if (result.getStatus() == Result.Status.OK) {
```

Comment:
no need to check, the SinglgeGfshCommand will check this, only when status is OK, it will go ahead and use the config object to update the cluster config.